### PR TITLE
fix(cli/program_state): Check for inline source maps before external ones

### LINKED ISSUE #6965: Main thread panic at Error throw
I'm experiencing a strange bug with `Deno`.

When a exception is thrown the main process panics. This is the trace:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', cli\fmt_errors.rs:86:8
stack backtrace:
   0:     0x7ff7af88a45b - <unknown>
   1:     0x7ff7af53cecc - <unknown>
   2:     0x7ff7af889483 - <unknown>
   3:     0x7ff7af888bb8 - <unknown>
   4:     0x7ff7af887d2e - <unknown>
   5:     0x7ff7af88efe5 - <unknown>
   6:     0x7ff7af53a4b0 - <unknown>
   7:     0x7ff7af53a2cc - <unknown>
   8:     0x7ff7af36cbf2 - <unknown>
   9:     0x7ff7af48ada8 - <unknown>
  10:     0x7ff7af48b345 - <unknown>
  11:     0x7ff7af53cecc - <unknown>
  12:     0x7ff7af751885 - <unknown>
  13:     0x7ff7af349c1d - <unknown>
  14:     0x7ff7af4daf6f - <unknown>
  15:     0x7ff7af8aa3f6 - <unknown>
  16:     0x7ff7af4ea67e - <unknown>
  17:     0x7ff7b048bf10 - CrashForExceptionInNonABICompliantCodeRange
  18:     0x7ff8e9777bd4 - BaseThreadInitThunk
  19:     0x7ff8ea16ce51 - RtlUserThreadStart
```

I don't know how to reproduce the bug but it happened in my project so I tried to isolate it in a separated branch:
https://github.com/AndreJesusBrito/WebAssembly-Language-Compiler/tree/rust-exception-bug-isolation

The bug happens in file `src/SyntaxAnalysis/SyntaxRule.ts` at line 25 when I throw an exception.
```ts
...
public getDerivation(terminal: string): RuleDerivation {
  const derivation = this.derivations.get(terminal);
  if (derivation) {
    return {
      derivationSymbols: [...derivation.derivationSymbols].reverse(),
      actions: derivation.actions,
    };
  } else {
    throw SyntaxError("Unexpected token '" + terminal + "' at rule " + this._name);            // THIS IS LINE 25
  }
}

public setDerivation(rule: SyntaxSymbol[], actions: ActionObj[], ...terminals: string[]): void {
  for (const terminal of terminals) {
    if (this.derivations.get(terminal) instanceof Array) {
      console.warn("'" + terminal + "' repeated terminal for rule " + this._name);
...
```

Even though I'm using a SyntaxError (that is not a very good idea and will change later) it seems to happen for any error including the base Error class.

The most strange thing to me is that by changing the white space on this file it can work correctly.
For example, when I remove line 28 it works fine:
```
error: Uncaught SyntaxError: Unexpected token '' at rule none
  public setDerivation(rule: SyntaxSymbol[], actions: ActionObj[], ...terminals: string[]): void {
            ^
    at SyntaxRule.getDerivation (file:///C:/Users/Andr%C3%A9%20Brito/Documents/GitHub/WebAssembly-Language-Compiler/src/SyntaxAnalysis/SyntaxRule.ts:28:13)
    at file:///C:/Users/Andr%C3%A9%20Brito/Documents/GitHub/WebAssembly-Language-Compiler/src/main.ts:4:1
```

I'm running using the command:
```
deno run src\main.ts
```

and my specs are:
```
deno 1.2.2
v8 8.5.216
typescript 3.9.2
OS: Windows_NT x64 10.0.18362
```

Hope this helps! :)

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes #6965.

I don't know if or where external source maps are still used, I guess that cleanup would be part of @kitsonk's TODOs here.

### CODE REVIEW

**Comment by nayeemrmn** [comment]
@bartlomieju PTAL

**Review by bartlomieju — COMMENTED** [review]
LGTM, let's wait for @kitsonk approval before landing

**Review by kitsonk — APPROVED** [review]
LGTM

Should have never been that way, thanks for fixing it @nayeemrmn.

**Pull request merged**
